src/query/query_executor.c

```diff
@@ -4719,7 +4719,8 @@ qexec_groupby (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_stat
     result = (gbstate.state == NO_ERROR || gbstate.state == SORT_PUT_STOP) ? NO_ERROR : ER_FAILED;
 
     /* check merge result */
-    if (XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY) && list_id->tuple_cnt != gbstate.input_recs)
+    if (gbstate.state == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY)
```

Comment:
`gbstate.state == SORT_PUT_STOP` seems not to be considered as an error too.

```suggestion
    if (result == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY)
```

Reply:
thanks